--- a/builddate.c	2024-02-26 19:53:26.000000000 +0100
+++ b/builddate.c	2024-03-31 16:45:28.549750267 +0200
@@ -1,18 +1,2 @@
-#define STRINGIFY2(arg) #arg
-#define STRINGIFY(arg) STRINGIFY2(arg)
-
 extern const char *buildstring;
-const char *buildstring =
-#ifdef VCREVISION
-STRINGIFY(VCREVISION)
-#else
-"-"
-#endif
-#ifndef NO_BUILD_TIMESTAMPS
-//" " __TIME__
-" " __DATE__
-#endif
-#ifdef BUILDTYPE
-" " STRINGIFY(BUILDTYPE)
-#endif
-;
+const char *buildstring = "%{PVR}";
